chore: add BPDM to umbrella chart

gomezbc opened the pull request:
<!-- 
Thanks for your contribution! 
Please follow the instructions on your PRs title and description.
aligned title description: '(feat|fix|chore|doc): _description of introduced change_'
Important: Contributing Guidelines can be found here: https://eclipse-tractusx.github.io/docs/oss/how-to-contribute
Info: <!- text comments ->  will be hidden from the rendered preview of your PR.
-->

## Description
<!-- 
Please describe your PR: 
- What does this PR introduce? 
- Does it fix a bug? 
- Does it add a new feature?
- Is it enhancing documentation?
-->

This PR adds BPDM chart to the umbrella repository. Due to incompatible Portal version, I had to bump portal chart to 2.0.0. As mentioned in this issue, https://github.com/eclipse-tractusx/portal/issues/248, in version 1.8.1 of Portal, it calls this enpoint `/companies/test-company/v6/` of the gate, which is incompatible with the added gate.

This BPDM is prepared to work with centralidp, but as it is planned for future release https://github.com/eclipse-tractusx/sig-release/issues/751, currently it will return 403 error in BPN step.

https://github.com/eclipse-tractusx/sig-release/issues/710

<!-- Please tag the related issue `Fixes or Updates #issue_number`, if applicable. -->

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

gomezbc commented:
@evegufy

evegufy commented on charts/umbrella/Chart.yaml:
```diff
@@ -36,7 +36,7 @@ dependencies:
     name: portal
     repository: https://eclipse-tractusx.github.io/charts/dev
-    version: 1.8.1
+    version: 2.0.0
```
Could you please restrict your changes to bpdm?
I know it won't work as a whole as long as portal isn't upgrade as well  (that's why we work on a separate upgrade branch until all components are ready). I'd upgrade portal, centraldip and sharedidp all together in a separate PR.

gomezbc replied:
Understood. I'll revert portal version upgrade. I'm ready to help with the portal, centraldip, and sharedidp upgrades if needed.

gomezbc replied:
Sure! In that case, should I create a separate pull request for each component, or do I upgrade them in one?

evegufy replied:
I think one pull request with the upgrade of all 3 components would be also good :)

evegufy reviewed (changes requested):
Hi @gomezbc thank you so much for the contribution!

evegufy commented on charts/umbrella/Chart.yaml:
```diff
@@ -72,6 +72,11 @@ dependencies:
     name: semantic-hub
     repository: https://eclipse-tractusx.github.io/charts/dev
     version: 0.2.2
+  # Bussines Partner Data Management
```
small typo in Business

evegufy commented on charts/umbrella/values.yaml:
```diff
@@ -681,6 +681,148 @@ managed-identity-wallet:
       nginx.ingress.kubernetes.io/enable-cors: "true"
 
+bpdm:
+  enabled: true
```
by default we add all the chart with enabled set to false

for testing we have 2 different values files, could you please add and enable bpdm in the one for the shared services? https://github.com/eclipse-tractusx/tractus-x-umbrella/blob/b6d50476ab4689b64e4b6418b538f9669407b9eb/charts/values-test-shared-services.yaml

evegufy reviewed (changes requested):
could you pleas also raise the chart version to 0.17.0? (the helm test workflow requires the increment to run https://github.com/eclipse-tractusx/tractus-x-umbrella/actions/runs/9844462507/job/27182944379?pr=106)

evegufy commented:
@gomezbc the review is currently blocked by the merged of https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/107, after it has been merged it would be great if you could rebase to the upgrade/24.05 branch

evegufy commented:
> @gomezbc the review is currently blocked by the merged of #107, after it has been merged it would be great if you could rebase to the upgrade/24.05 branch

@gomezbc the pr #107 has been merged now, could you please rebase? then the helm test workflow should pass the check for chart version bump

gomezbc commented:
@evegufy Done 👍🏼

evegufy reviewed (changes requested):
Hi @gomezbc the helm test workflow did its job correctly now :) and uncovered some [linting issues,](https://github.com/eclipse-tractusx/tractus-x-umbrella/actions/runs/9870098659/job/27280181316?pr=106#step:13:58) could you please fix them?

gomezbc commented:
@evegufy I have resolved the linting issues. I noticed that other components disable PostgreSQL persistence by default, so I have done the same for the BPDM database.

evegufy reviewed (changes requested):
one linting issue still:
> charts/umbrella/values.yaml
>  Error: 748:19 [trailing-spaces] trailing spaces

evegufy reviewed (commented):
Hi @gomezbc sorry for the delay, I finally came around to reviewing again.
When install I'm facing 401 errors for pool, gate and dummy:
[logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log)
[logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log)
[logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)

This might be due to missing the bdpm config discussed here https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297

@nicoprow has opened https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/108 which might contains the config.

evegufy reviewed (commented):
Hi @gomezbc sorry for the delay, I finally came around to reviewing again.
When install I'm facing 401 errors for pool, gate and dummy:
[logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log)
[logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log)
[logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)

This might be due to missing the bdpm config discussed here https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297

@nicoprow has opened https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/108 which might contains the config.

nicoprow commented:
> Hi @gomezbc sorry for the delay, I finally came around to reviewing again. When install I'm facing 401 errors for pool, gate and dummy: [logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log) [logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log) [logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)
> 
> This might be due to missing the bdpm config discussed here [eclipse-tractusx/portal-backend#819 (comment)](https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297)
> 
> @nicoprow has opened #108 which might contains the config.

@gomezbc:

As @evegufy already said, you can use the newest hotfix version for BPDM 5.0.3 with the configuration provided in #108. The files also contain README instructions and a workflow to test the BPDM chart dependency setup.

The new version has the advantage that the postgres does not need to have a fullnameOverride anymore. 

The ingress can be changed for BPDM but I experienced compatibility problems with the Portal. At least for the Partner Network page, I discovered that that Portal version has outdated endpoint URLs when trying to access the Pool. I don't see a way to fix that except upgrading the Portal so the correct endpoint paths are used.

I did not try the registration workflow with the Portal, so I don't know which path to the BPDM Gate the Portal expects there.

nicoprow commented on charts/umbrella/README.md:
```diff
@@ -123,6 +123,10 @@ If you still face DNS issues, add the hosts to your /etc/hosts file:
+192.168.49.2    partners-pool.tx.test
+192.168.49.2    partners-gate.tx.test
+192.168.49.2    partners-orchestrator.tx.test
+192.168.49.2    partners-cleaning.tx.test
```
The cleaning service does not have a API and does not need to be exposed. Have a look here to see the recommended settings: #108

nicoprow commented on charts/umbrella/values.yaml:
```diff
@@ -681,6 +681,149 @@ managed-identity-wallet:
+          paths:
+            - path: /
+              pathType: Prefix
+    springProfiles:
```
not needed anymore, the default configuration already enbales auth. see: https://github.com/eclipse-tractusx/bpdm/blob/release/6.0.x/INSTALL.md

nicoprow commented on charts/umbrella/values.yaml:
```diff
@@ -681,6 +681,149 @@ managed-identity-wallet:
       nginx.ingress.kubernetes.io/cors-allow-methods: GET
       nginx.ingress.kubernetes.io/enable-cors: "true"
 
+bpdm:
```
bpdm-gate, bpdm-pool and bpdm-orchestrator expect default permissions that are different from the Central-IDP version in the umbrella chart. Therefore the default permissions have to be overwritten like seen here #108

gomezbc commented:
Hi @nicoprow, thanks for the comments. I have review your configuration and is awesome. As you have a better understanding of BPDM, I think the best approach would be to use your configuration.

Regarding the compatibility with the portal, in my local setup, I had to upgrade to version 2.0.0 to make it compatible with BPDM. Using that version, the registration process works correctly.

As I discussed with @evegufy (https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/106#discussion_r1669237002), I will provide an upgrade to the portal, centralidp, and sharedidp. I will try to provide it as soon as I can.

gomezbc commented:
Hi @nicoprow

After creating the PR for the portal version update https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/110, I switched my configuration to yours to connect the portal with BPDM. However, I encountered some permission issues. Upon comparing the roles required by BPDM with those defined in CentralIdp, I resolved these issues.

The roles you defined for the Gate, Pool, and Orchestrator do not match those used by the CentralIdp sa-cl7-cx-5 client.

For example, here are the roles you set for Gate (Cl16-CX-BPDMGate):
```yaml
permissions:
  readInputPartner: view_company_data
  writeInputPartner: update_company_data
  readOutputPartner: view_shared_data
  readInputChangelog: view_company_data
  readOutputChangelog: view_shared_data
  readSharingState: view_company_data
  writeSharingState: update_company_data
  read_stats: view_company_data
```
And here are the roles provided by CentralIdp:

![image](https://github.com/user-attachments/assets/cedc550f-d3de-43b9-8614-253040d94c11)

To fix the Gate and Pool permissions, I deleted those settings, as the default configuration matches Cl16-CX-BPDMGate and Cl17-CX-BPDM.

For the Orchestrator, I modified the values you set in permissions to these:
```yaml
permissions:
  createTask: "write_partner"
  readTask: "write_partner"
  reservation:
    clean: "write_partner"
    cleanAndSync: "write_partner"
    poolSync: "write_partner"
  result:
    clean: "write_partner"
    cleanAndSync: "write_partner"
    poolSync: "write_partner"
```

> [!warning]
> I have successfully completed the BPN creation process from the portal using the updated portal version [here](https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/110). It will not be possible to complete the BPN creation process with the portal version defined in this PR.

evegufy commented:
> > Hi @gomezbc sorry for the delay, I finally came around to reviewing again. When install I'm facing 401 errors for pool, gate and dummy: [logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log) [logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log) [logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)
> > This might be due to missing the bdpm config discussed here [eclipse-tractusx/portal-backend#819 (comment)](https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297)
> > @nicoprow has opened #108 which might contains the config.
> 
> @gomezbc:
> 
> As @evegufy already said, you can use the newest hotfix version for BPDM 5.0.3 with the configuration provided in #108. The files also contain README instructions and a workflow to test the BPDM chart dependency setup.
> 
> The new version has the advantage that the postgres does not need to have a fullnameOverride anymore.
> 
> The ingress can be changed for BPDM but I experienced compatibility problems with the Portal. At least for the Partner Network page, I discovered that that Portal version has outdated endpoint URLs when trying to access the Pool. I don't see a way to fix that except upgrading the Portal so the correct endpoint paths are used.
> 
> I did not try the registration workflow with the Portal, so I don't know which path to the BPDM Gate the Portal expects there.

Just quickly clarifying on what you mentioned regarding outdated endpoints in the portal: I assume you tested with the version of the portal which is currently still in the main branch of the umbrella repo, which is the portal version 1.8.0 for the release 24.03, that one is expected to be not compatible with the BPDM version for 24.05, the portal version for the 24.05 release is the 2.0.0.
Overview of compatible version for the for R24.05 https://eclipse-tractusx.github.io/CHANGELOG#2405---2024-05-29

nicoprow commented:
> Hi @nicoprow
> 
> After creating the PR for the portal version update #110, I switched my configuration to yours to connect the portal with BPDM. However, I encountered some permission issues. Upon comparing the roles required by BPDM with those defined in CentralIdp, I resolved these issues.
> 
> The roles you defined for the Gate, Pool, and Orchestrator do not match those used by the CentralIdp sa-cl7-cx-5 client.
> 
> For example, here are the roles you set for Gate (Cl16-CX-BPDMGate):
> 
> ```yaml
> permissions:
>   readInputPartner: view_company_data
>   writeInputPartner: update_company_data
>   readOutputPartner: view_shared_data
>   readInputChangelog: view_company_data
>   readOutputChangelog: view_shared_data
>   readSharingState: view_company_data
>   writeSharingState: update_company_data
>   read_stats: view_company_data
> ```
> 
> And here are the roles provided by CentralIdp:
> 
> ![image](https://private-user-images.githubusercontent.com/77118356/350456343-cedc550f-d3de-43b9-8614-253040d94c11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2MTc2NDAsIm5iZiI6MTcyMTYxNzM0MCwicGF0aCI6Ii83NzExODM1Ni8zNTA0NTYzNDMtY2VkYzU1MGYtZDNkZS00M2I5LTg2MTQtMjUzMDQwZDk0YzExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIyVDAzMDIyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllMjk2NTllMThjM2I3MTI1ZGExMDA4YzQyZjE0YWNiZGU5NWVkM2MwMTNlMDc5MjhiOTM1MmMwZjA5NjNmZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IqZnplp2y9eL4MJ_qOZqg60PRTWiVkK6oxQttmCfvTI)
> 
> To fix the Gate and Pool permissions, I deleted those settings, as the default configuration matches Cl16-CX-BPDMGate and Cl17-CX-BPDM.
> 
> For the Orchestrator, I modified the values you set in permissions to these:
> 
> ```yaml
> permissions:
>   createTask: "write_partner"
>   readTask: "write_partner"
>   reservation:
>     clean: "write_partner"
>     cleanAndSync: "write_partner"
>     poolSync: "write_partner"
>   result:
>     clean: "write_partner"
>     cleanAndSync: "write_partner"
>     poolSync: "write_partner"
> ```
> 
> Warning
> 
> I have successfully completed the BPN creation process from the portal using the updated portal version [here](https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/110). It will not be possible to complete the BPN creation process with the portal version defined in this PR.

Yea, your permission configuration is exactly right to do for the Central-IDP 24.05 release. Great that you made it work!

If you want you can remove the base-URL overwrite for the BPDM clients as they are able to find each other by service name. I don't think is necessary to change it but I thought I should mention it (see the comment below)

nicoprow commented on charts/umbrella/values.yaml:
```diff
@@ -681,6 +681,200 @@ managed-identity-wallet:
+        # We will reuse the general BPDM admin technical user to establish connection between the services
+        client:
+          pool:
+            base-url: "http://business-partners.tx.test/pool"
```
On default BPDM services try to find each other by their release name in the cluster. Since the services are created by the same Helm release and are running in the same cluster/namespace we can remove the base-url overwrite here.

nicoprow commented on charts/umbrella/values.yaml:
```diff
@@ -681,6 +681,200 @@ managed-identity-wallet:
+            registration:
+              client-id: *bpdmAdminClientId
+          orchestrator:
+            base-url: "http://business-partners.tx.test/orchestrator"
```
Same here

nicoprow commented on charts/umbrella/values.yaml:
```diff
@@ -681,6 +681,200 @@ managed-identity-wallet:
+        # We will reuse the general BPDM admin technical user to establish connection between the services
+        client:
+          orchestrator:
+            base-url: "http://business-partners.tx.test/orchestrator"
```
Same here

nicoprow commented on charts/umbrella/values.yaml:
```diff
@@ -681,6 +681,200 @@ managed-identity-wallet:
+        # We reuse the general BPDM admin technical user to establish connection between the services
+        client:
+          orchestrator:
+            base-url: "http://business-partners.tx.test/orchestrator"
```
Same here

nicoprow reviewed (commented):
Comments below as an option to further simplify the configuration.

Since there is now also the values-test-bpdm.yaml inside I would also expect the BPDM chart test to be part of this pull request.

evegufy commented:
@evegufy will check out the chart testing

evegufy commented:
Hi @gomezbc https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/113 has been merged, could you please add bpdm to the the 1rst test file?

gomezbc commented:
> Hi @gomezbc #113 has been merged, could you please add bpdm to the the 1rst test file?

Done! 😀

evegufy commented:
> > Hi @gomezbc #113 has been merged, could you please add bpdm to the the 1rst test file?
> 
> Done! 😀

Awesome :)
